client/SDL/SDL2/sdl_freerdp.cpp

```diff
@@ -1190,9 +1237,13 @@ static DWORD WINAPI sdl_client_thread_proc(SdlContext* sdl)
 	BOOL rc = freerdp_connect(instance);
 
 	rdpContext* context = sdl->context();
+	WINPR_ASSERT(context);
+
 	rdpSettings* settings = context->settings;
 	WINPR_ASSERT(settings);
 
+	ScopeGuard guard([&]() { sdl_client_cleanup(sdl, exit_code, error_msg); });
```

Comment:
@hardening hmm, this here might indeed need another refactoring.
thinking about it it would be a lot better to encapsulate the stuff in a function and call them one after the other.
e.g. the whole code doing `freerdp_connect` in one, `while (!freerdp_shall_disconnect_context(context))` in another and then before returning calling `sdl_client_cleanup`

so no need for a `ScopeGuard` any more and even better readable than before.